kvserver: enable v1 flow control integration tests under duress

Disabled in https://github.com/cockroachdb/cockroach/pull/132125 -- extended CI failures for V1 tests (not V2). The cause is usually identical, streams disconnecting when the test doesn't expect it.


<details><summary>Details</summary>
<p>


Disabled in https://github.com/cockroachdb/cockroach/pull/132121 and https://github.com/cockroachdb/cockroach/pull/132374 to unblock merging.


https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelExtendedCi/17247400?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildChangesSection=true&expandBuildProblemsSection=true&expandBuildTestsSection=true

```
Failed
=== RUN   TestFlowControlRaftSnapshot
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestFlowControlRaftSnapshot952225615
    test_log_scope.go:76: use -show-logs to present logs inline
    datadriven.go:144:
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3726/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/kvserver_test_/kvserver_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/testdata/flow_control_integration/raft_snapshot:1:
        output didn't match expected:
        @@ -159,15 +159,10 @@
         SELECT range_id, store_id, crdb_internal.humanize_bytes(total_tracked_tokens::INT8)
            FROM crdb_internal.kv_flow_control_handles
           range_id | store_id | total_tracked_tokens
         -----------+----------+-----------------------
        -  70       | 1        | 0 B
        -  70       | 2        | 0 B
        -  70       | 3        | 0 B
        -  70       | 4        | 0 B
        -  70       | 5        | 0 B
         -- Another view of tokens, using /inspectz-backed vtables.
         SELECT store_id,
              crdb_internal.humanize_bytes(available_regular_tokens),
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestFlowControlRaftSnapshot952225615
--- FAIL: TestFlowControlRaftSnapshot (76.80s)
```

</p>
</details> 


Jira issue: CRDB-42941

Epic CRDB-37515

--- a/pkg/kv/kvserver/flow_control_integration_test.go
+++ b/pkg/kv/kvserver/flow_control_integration_test.go
@@ -68,9 +68,6 @@ import (
 func TestFlowControlBasic(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-	// TOOD(kvoli,pav-kv,sumeerbhola): Enable this test under all conditions
-	// after fixing the flakiness introduced by #132121.
-	skip.UnderDuressWithIssue(t, 132310)
 
 	testutils.RunTrueAndFalse(t, "always-enqueue", func(t *testing.T, alwaysEnqueue bool) {
 		ctx := context.Background()
@@ -223,9 +220,6 @@ ORDER BY streams DESC;
 func TestFlowControlRangeSplitMerge(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-	// TOOD(kvoli,pav-kv,sumeerbhola): Enable this test under all conditions
-	// after fixing the flakiness introduced by #132121.
-	skip.UnderDuressWithIssue(t, 132310)
 
 	ctx := context.Background()
 	const numNodes = 3
@@ -341,9 +335,6 @@ ORDER BY streams DESC;
 func TestFlowControlBlockedAdmission(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-	// TOOD(kvoli,pav-kv,sumeerbhola): Enable this test under all conditions
-	// after fixing the flakiness introduced by #132121.
-	skip.UnderDuressWithIssue(t, 132310)
 
 	ctx := context.Background()
 	const numNodes = 3
@@ -455,9 +446,6 @@ ORDER BY name ASC;
 func TestFlowControlAdmissionPostSplitMerge(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-	// TOOD(kvoli,pav-kv,sumeerbhola): Enable this test under all conditions
-	// after fixing the flakiness introduced by #132121.
-	skip.UnderDuressWithIssue(t, 132310)
 
 	ctx := context.Background()
 	const numNodes = 3
@@ -594,9 +582,6 @@ ORDER BY streams DESC;
 func TestFlowControlCrashedNode(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-	// TOOD(kvoli,pav-kv,sumeerbhola): Enable this test under all conditions
-	// after fixing the flakiness introduced by #132121.
-	skip.UnderDuressWithIssue(t, 132310)
 
 	ctx := context.Background()
 	const numNodes = 2
@@ -714,9 +699,6 @@ func TestFlowControlCrashedNode(t *testing.T) {
 func TestFlowControlRaftSnapshot(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-	// TOOD(kvoli,pav-kv,sumeerbhola): Enable this test under all conditions
-	// after fixing the flakiness introduced by #132121.
-	skip.UnderDuressWithIssue(t, 132310)
 
 	const numServers int = 5
 	stickyServerArgs := make(map[int]base.TestServerArgs)
@@ -1014,9 +996,6 @@ SELECT store_id,
 func TestFlowControlRaftTransportBreak(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-	// TOOD(kvoli,pav-kv,sumeerbhola): Enable this test under all conditions
-	// after fixing the flakiness introduced by #132121.
-	skip.UnderDuressWithIssue(t, 132310)
 
 	ctx := context.Background()
 	const numNodes = 3
@@ -1129,9 +1108,6 @@ func TestFlowControlRaftTransportBreak(t *testing.T) {
 func TestFlowControlRaftTransportCulled(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-	// TOOD(kvoli,pav-kv,sumeerbhola): Enable this test under all conditions
-	// after fixing the flakiness introduced by #132121.
-	skip.UnderDuressWithIssue(t, 132310)
 
 	ctx := context.Background()
 	const numNodes = 3
@@ -1268,9 +1244,6 @@ func TestFlowControlRaftTransportCulled(t *testing.T) {
 func TestFlowControlRaftMembership(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-	// TOOD(kvoli,pav-kv,sumeerbhola): Enable this test under all conditions
-	// after fixing the flakiness introduced by #132121.
-	skip.UnderDuressWithIssue(t, 132310)
 
 	ctx := context.Background()
 	st := cluster.MakeTestingClusterSettings()
@@ -1401,9 +1374,6 @@ ORDER BY name ASC;
 func TestFlowControlRaftMembershipRemoveSelf(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-	// TOOD(kvoli,pav-kv,sumeerbhola): Enable this test under all conditions
-	// after fixing the flakiness introduced by #132121.
-	skip.UnderDuressWithIssue(t, 132310)
 
 	testutils.RunTrueAndFalse(t, "transfer-lease-first", func(t *testing.T, transferLeaseFirst bool) {
 		ctx := context.Background()
@@ -1532,9 +1502,6 @@ ORDER BY name ASC;
 func TestFlowControlClassPrioritization(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-	// TOOD(kvoli,pav-kv,sumeerbhola): Enable this test under all conditions
-	// after fixing the flakiness introduced by #132121.
-	skip.UnderDuressWithIssue(t, 132310)
 
 	ctx := context.Background()
 	const numNodes = 5
@@ -1618,9 +1585,6 @@ func TestFlowControlClassPrioritization(t *testing.T) {
 func TestFlowControlQuiescedRange(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-	// TOOD(kvoli,pav-kv,sumeerbhola): Enable this test under all conditions
-	// after fixing the flakiness introduced by #132121.
-	skip.UnderDuressWithIssue(t, 132310)
 
 	ctx := context.Background()
 	const numNodes = 3
@@ -1761,9 +1725,6 @@ ORDER BY name ASC;
 func TestFlowControlUnquiescedRange(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-	// TOOD(kvoli,pav-kv,sumeerbhola): Enable this test under all conditions
-	// after fixing the flakiness introduced by #132121.
-	skip.UnderDuressWithIssue(t, 132310)
 
 	ctx := context.Background()
 	const numNodes = 3
@@ -1917,9 +1878,6 @@ ORDER BY name ASC;
 func TestFlowControlTransferLease(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-	// TOOD(kvoli,pav-kv,sumeerbhola): Enable this test under all conditions
-	// after fixing the flakiness introduced by #132121.
-	skip.UnderDuressWithIssue(t, 132310)
 
 	ctx := context.Background()
 	const numNodes = 5
@@ -2008,9 +1966,6 @@ ORDER BY name ASC;
 func TestFlowControlLeaderNotLeaseholder(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-	// TOOD(kvoli,pav-kv,sumeerbhola): Enable this test under all conditions
-	// after fixing the flakiness introduced by #132121.
-	skip.UnderDuressWithIssue(t, 132310)
 
 	ctx := context.Background()
 	const numNodes = 5
@@ -2144,9 +2099,6 @@ ORDER BY name ASC;
 func TestFlowControlGranterAdmitOneByOne(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-	// TOOD(kvoli,pav-kv,sumeerbhola): Enable this test under all conditions
-	// after fixing the flakiness introduced by #132121.
-	skip.UnderDuressWithIssue(t, 132310)
 
 	ctx := context.Background()
 	const numNodes = 3

--- a/pkg/kv/kvserver/store_rebalancer.go
+++ b/pkg/kv/kvserver/store_rebalancer.go
@@ -149,6 +149,7 @@ type StoreRebalancer struct {
 	processTimeoutFn        func(replica CandidateReplica) time.Duration
 	objectiveProvider       RebalanceObjectiveProvider
 	subscribedToSpanConfigs func() bool
+	disabled                func() bool
 }
 
 // NewStoreRebalancer creates a StoreRebalancer to work in tandem with the
@@ -190,6 +191,10 @@ func NewStoreRebalancer(
 				return false
 			}
 			return !rq.store.cfg.SpanConfigSubscriber.LastUpdated().IsEmpty()
+		},
+		disabled: func() bool {
+			return LoadBasedRebalancingMode.Get(&st.SV) == LBRebalancingOff ||
+				rq.store.cfg.TestingKnobs.DisableStoreRebalancer
 		},
 	}
 	sr.AddLogTag("store-rebalancer", nil)
@@ -308,15 +313,13 @@ func (sr *StoreRebalancer) Start(ctx context.Context, stopper *stop.Stopper) {
 				timer.Read = true
 				timer.Reset(jitteredInterval(allocator.LoadBasedRebalanceInterval.Get(&sr.st.SV)))
 			}
-
+			if sr.disabled() {
+				continue
+			}
 			// Once the rebalance mode and rebalance objective are defined for
 			// this loop, they are immutable and do not change. This avoids
 			// inconsistency where the rebalance objective changes and very
 			// different or contradicting actions are then taken.
-			mode := sr.RebalanceMode()
-			if mode == LBRebalancingOff {
-				continue
-			}
 			if !sr.subscribedToSpanConfigs() {
 				continue
 			}
@@ -326,7 +329,7 @@ func (sr *StoreRebalancer) Start(ctx context.Context, stopper *stop.Stopper) {
 
 			hottestRanges := sr.replicaRankings.TopLoad(objective.ToDimension())
 			options := sr.scorerOptions(ctx, objective.ToDimension())
-			rctx := sr.NewRebalanceContext(ctx, options, hottestRanges, mode)
+			rctx := sr.NewRebalanceContext(ctx, options, hottestRanges, sr.RebalanceMode())
 			sr.rebalanceStore(ctx, rctx)
 		}
 	})

--- a/pkg/kv/kvserver/testing_knobs.go
+++ b/pkg/kv/kvserver/testing_knobs.go
@@ -176,6 +176,9 @@ type StoreTestingKnobs struct {
 	DisableReplicaGCQueue bool
 	// DisableReplicateQueue disables the replication queue.
 	DisableReplicateQueue bool
+	// DisableStoreRebalancer turns off the store rebalancer which moves replicas
+	// and leases.
+	DisableStoreRebalancer bool
 	// DisableLoadBasedSplitting turns off LBS so no splits happen because of load.
 	DisableLoadBasedSplitting bool
 	// LoadBasedSplittingOverrideKey returns a key which should be used for load

--- a/pkg/testutils/testcluster/testcluster.go
+++ b/pkg/testutils/testcluster/testcluster.go
@@ -581,6 +581,7 @@ func (tc *TestCluster) AddServer(
 		stkCopy.DisableSplitQueue = true
 		stkCopy.DisableMergeQueue = true
 		stkCopy.DisableReplicateQueue = true
+		stkCopy.DisableStoreRebalancer = true
 		serverArgs.Knobs.Store = &stkCopy
 	}
 

